kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/QueryGatewayExtensions.kt

```diff
@@ -106,3 +108,114 @@ inline fun <reified R, reified Q> QueryGateway.queryForOptional(query: Q): Compl
+ * @return [Stream] a stream of results
+ * @see QueryGateway.scatterGather
+ * @see ResponseTypes
+ * @since 0.2.0
+ */
+inline fun <reified R, reified Q> QueryGateway.scatterGatherForSingle(query: Q, timeout: Long,
+                                                                      timeUnit: TimeUnit): Stream<R> {
+    return this.scatterGather(query, ResponseTypes.instanceOf(R::class.java), timeout, timeUnit)
+}
+
+/**
+ * Reified version of [QueryGateway.scatterGather] with explicit query name
+ * which expects an Stream object as a response using [org.axonframework.messaging.responsetypes.InstanceResponseType]
+ * @param query Query to send
+ * @param queryName Name of the query
+ * @param timeout a timeout for the query
+ * @param timeUnit the selected TimeUnit for the given timeout
+ * @param [Q] the type of payload of the query
+ * @param [R] the response class contained in the given responseType
+ * @return [Stream] a stream of results
+ * @see QueryGateway.scatterGather
+ * @see ResponseTypes
+ * @since 0.2.0
+ */
+inline fun <reified R, reified Q> QueryGateway.scatterGatherForSingle(queryName: String, query: Q, timeout: Long,
+                                                                      timeUnit: TimeUnit): Stream<R> {
+    return this.scatterGather(queryName, query, ResponseTypes.instanceOf(R::class.java), timeout, timeUnit)
+}
+
+/**
+ * Reified version of [QueryGateway.scatterGather]
+ * which expects a collection as a response using [org.axonframework.messaging.responsetypes.MultipleInstancesResponseType]
+ * @param query Query to send
+ * @param timeout a timeout for the query
+ * @param timeUnit the selected TimeUnit for the given timeout
+ * @param [Q] the type of payload of the query
+ * @param [R] the response class contained in the given responseType
+ * @return [Stream] a stream of results
+ * @see QueryGateway.scatterGather
+ * @see ResponseTypes
+ * @since 0.2.0
+ */
+inline fun <reified R, reified Q> QueryGateway.scatterGatherForMultiple(query: Q, timeout: Long,
+                                                                        timeUnit: TimeUnit): Stream<List<R>> {
+    return this.scatterGather(query, ResponseTypes.multipleInstancesOf(R::class.java), timeout, timeUnit)
+}
+
+/**
+ * Reified version of [QueryGateway.scatterGather] with explicit query name
+ * which expects a collection as a response using [org.axonframework.messaging.responsetypes.MultipleInstancesResponseType]
+ * @param query Query to send
+ * @param queryName Name of the query
+ * @param timeout a timeout for the query
+ * @param timeUnit the selected TimeUnit for the given timeout
+ * @param [Q] the type of payload of the query
+ * @param [R] the response class contained in the given responseType
+ * @return [Stream] a stream of results
+ * @see QueryGateway.scatterGather
+ * @see ResponseTypes
+ * @since 0.2.0
+ */
+inline fun <reified R, reified Q> QueryGateway.scatterGatherForMultiple(queryName: String, query: Q, timeout: Long,
```

Comment:
The new convention is `scatterGatherMany`, without `For` and `Many` instead of `Multiple`